File: app/src/main/java/com/bot/man/model/Repository.java
package com.bot.man.model;

import com.bot.man.api.ApiHandler;
import com.bot.man.viewmodel.MainViewModel;

public class Repository {
	private static Repository mRepository;

	private Repository() {
	}

	public static Repository getInstance() {
		if (mRepository == null) {
			mRepository = new Repository();
		}
		return mRepository;
	}

	public void fetchAPI(
			MainViewModel.OnResultListener onResultListener, String API_KEY) {
		ApiHandler.getInstance().fetchData(onResultListener, API_KEY);
	}

}

File: app/src/main/java/com/bot/man/model/Repository.kt
package com.bot.man.model

import com.bot.man.api.ApiHandler
import com.bot.man.viewmodel.MainViewModel.OnResultListener

class Repository private constructor() {
    fun fetchAPI(
            onResultListener: OnResultListener, API_KEY: String) {
        ApiHandler.instance?.fetchData(onResultListener, API_KEY)
    }

    companion object {
        private var mRepository: Repository? = null

        @JvmStatic
        val instance: Repository?
            get() {
                if (mRepository == null) {
                    mRepository = Repository()
                }
                return mRepository
            }
    }
}